git/lib

```diff
@@ -2,19 +2,43 @@
 
 : "${BASH_LIB_DIR:?BASH_LIB_DIR must be set. Please source bash-lib/init before other scripts from bash-lib.}"
 
+
+function bl_git_available(){
+    type git >/dev/null || bl_fail "Git binary not found in ${PATH}"
```

Comment:
I'm sure this is fine, but for my understanding why `type` instead of `which` or `command`?

Reply:
The "not found" error message will leak out of all of these checks. I have no issue with that, just pointing it out in case it wasn't your intention.

Reply:
I'm using type here because I want to be able to replace binaries with dummy functions for testing.